types/match-sorter/index.d.ts

```diff
@@ -18,24 +19,15 @@ declare namespace matchSorter {
     }
 }
 
-interface MinRanking {
-    minRanking: number;
-    key: string;
-}
-
-interface MaxRanking {
-    maxRanking: number;
-    key: string;
-}
-
-interface MinMaxRanking {
-    minRanking: number;
-    maxRanking: number;
-    key: string;
+interface ExtendedKeyOptions<T> {
```

Comment:
How about something like this?

`type ExtendedKeyOptions<T> = { key: string | ((item: T) => string) } & ({ minRanking: number; } | { maxRanking: number; } | { treshold: number; });`

Reply:
Looks better, thanks. Though this would imply you couldn't use threshold together with maxRanking or minRanking. I think I have found the solution for this in my recent commit